gatorgrade/generate/generate.py

```diff
@@ -1,21 +1,82 @@
-"""Generate a GatorGrade configuration file.
+"""Generate a YAML file with default messages and specific paths."""
+import os
+from typing import List
 
-Gatorgrade file will include paths to whitelisted files
-and default GatorGrader checks.
-"""
+# Define colors for terminal output
+OKGREEN = "\033[92m"
+WARNING = "\033[93m"
+FAIL = "\033[91m"
 
-# Import the necessary libraries
 
-
-def create_targeted_paths_list():
+def create_targeted_paths_list(
+    key_word_list: List[str], relative_run_path: str = "."
+) -> List[str]:
     """Generate a list of targeted paths by walking the paths."""
+    targeted_paths = []
     # Go through the root repo, the sub dictionaries and files.
-    # Select only files in the dictionaries with specific names.
-    # Add those targeted file paths into a list and returns it.
+    # The os.walk will only scan the paths.
+    # So the empty folders containing nothing won't be gone through
+    for dirpath, _, filenames in os.walk(relative_run_path):
+        # Split path string into multiple layers of directories
+        path_dir_list = dirpath.split("/")
+        # Ignore folder starting with double underscore
+        if any(path.startswith("__") for path in path_dir_list):
+            continue
+        # Ignore hidden folders and first layer. the root repo is always dot
+        # Keep double dot. It represents the returning sign
+        if any(
+            path.startswith(".") and not path.startswith("..")
+            for path in path_dir_list[1:]
+        ):
+            continue
+        for filename in filenames:
+            # Ignore the file starting with double underscore and hidden file
+            if filename.startswith("__") or filename.startswith("."):
+                continue
+            # Add paths when they have the key words in the second and the third directories
+            # For the path with only two directories,
+            # check key words in the second directory folder name
+            if len(path_dir_list) == 2:
+                if path_dir_list[1] in key_word_list:
```

Comment:
What do you think about moving `key_word_list` to a dictionary?

My thought was then it would be constant-time lookup instead of pressumably `O(n)`. If you are interested in helping me, I was trying to find the python language specification for the `in` operator so that I could see how it is implemented. Not a big deal, only if you are interested 😄

Reply:
That's a GREAT idea. Thanks a lot. I am willing to do it but have to be on weekend. I am a little busy these two days